node/nodeutil/controller.go

```diff
@@ -0,0 +1,123 @@
+package nodeutil
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/virtual-kubelet/virtual-kubelet/node"
+)
+
+type ControllerManager struct {
+	NodeController *node.NodeController
+	PodController  *node.PodController
+
+	mu    sync.Mutex
+	ready chan struct{}
+	done  chan struct{}
+	err   error
+}
+
+func (c *ControllerManager) init() {
```

Comment:
Isn't this better done in a constructor New()? And hide the type from direct initialization>

Reply:
Yeah, originally was wanting to avoid a constructor, then realized I needed a mutex 🤷 oh well.
Updated to add constructor.